ui/lib/core/addon/components/replication-page.js

```diff
@@ -9,6 +11,26 @@ const MODE = {
 
 export default Component.extend({
   layout,
+  store: service(),
+  reindexingDetails: null,
+
+  didReceiveAttrs() {
+    this._super(arguments);
+    this.getReplicationModeStatus.perform();
+  },
+  getReplicationModeStatus: task(function*() {
+    let resp;
+    const { replicationMode } = this.model;
+    try {
+      resp = yield this.get('store')
+        .adapterFor('replication-mode')
+        .fetchStatus(replicationMode);
+    } catch (e) {
+      console.log(e);
```

Comment:
Do we want to set some sort of error on the page if we aren't able to get the status? Or is that even a realistic scenario?

Reply:
great question! i think we should be fine in this scenario because if we can't find the reindexing status we just wouldn't show the reindexing alert at the top of the page. the only reason i could see us getting an API error here is if the replication mode wasn't set, but i don't think that should ever happen 'cause these components are only used on pages where replication is already enabled & thus the mode is set. :)